lib/apiservers/service/swagger.json

```diff
@@ -0,0 +1,879 @@
+        "parameters": [
+          { "$ref": "#/parameters/target" },
+          { "$ref": "#/parameters/vch-id" },
+          { "$ref": "#/parameters/vch" },
+          { "$ref": "#/parameters/thumbprint" }
+        ],
+        "responses": {
+          "202": { "$ref": "#/responses/vsphere-task" },
+          "default": { "$ref": "#/responses/error" }
+        }
+      },
+      "patch": {
+        "summary": "Reconfigure the target VCH",
+        "description": "Making a `PATCH` request on a VCH resource (with a body as described in RFC 7396) will update a subset of that VCH's configuration. As `PATCH` is an explicit request to update a set of fields, fields which cannot be modified must not appear in the body of the `PATCH` request, even if the modification would be a no-op.",
+        "consumes": [
+          "application/merge-patch+json"
+        ],
+        "parameters": [
+          { "$ref": "#/parameters/target" },
+          { "$ref": "#/parameters/vch-id" },
+          { "$ref": "#/parameters/vch" },
+          { "$ref": "#/parameters/thumbprint" }
+        ],
+        "responses": {
+          "202": { "$ref": "#/responses/vsphere-task" },
+          "default": { "$ref": "#/responses/error" }
+        }
+      },
+      "post": {
+        "summary": "Perform an action on the target VCH",
+        "description": "Making a `POST` request on a VCH resource with an action of `upgrade` will initiate an upgrade of the VCH. The body of the request will be a JSON object containing the following optional properties: `bootstrap-iso` (a reference to a known bootstrap ISO on the OVA) and `rollback` (a boolean value).",
+        "parameters": [
+          { "$ref": "#/parameters/target" },
+          { "$ref": "#/parameters/vch-id" },
+          { "$ref": "#/parameters/vch-action" },
+          { "$ref": "#/parameters/thumbprint" }
+        ],
+        "responses": {
+          "202": { "$ref": "#/responses/vsphere-task" },
+          "default": { "$ref": "#/responses/error" }
+        }
+      },
+      "delete": {
+        "summary": "Delete the target VCH",
+        "description": "Making a `DELETE` request on a VCH resource will delete that VCH.",
+        "parameters": [
+          { "$ref": "#/parameters/target" },
+          { "$ref": "#/parameters/vch-id" },
+          { "$ref": "#/parameters/thumbprint" }
+        ],
+        "responses": {
+          "202": { "$ref": "#/responses/vsphere-task" },
+          "default": { "$ref": "#/responses/error" }
+        }
+      }
+    },
+    "/{target}/datacenter/{datacenter}": {
+      "get": {
+        "summary": "Show information about the specified vSphere resources",
+        "description": "Making a `GET` request on a datacenter will return information about the state of the host firewall on those resources.",
```

Comment:
The path of the resource `/{target}/datacenter/` does not indicate it has anything to do with firewalls. Can we change that reflect what the call is actually about?

Reply:
I expect that the resource will include additional information; firewall status is just the first item.

We could expose separate resources for each piece of information (`/{target}/datacenter/{datacenter}/firewall`, `/{target}/datacenter/{datacenter}/foo`, `/{target}/datacenter/{datacenter}/bar`), but instead I think it makes sense to have a single `/{target}/datacenter/{datacenter}` endpoint with a response body like below, and add sub-resources on an as-needed basis if necessary:
```json
{
    "firewall": { 
    },
    "foo": {
    },
    "bar": {
    }
}
```

Reply:
Yes, this is fine if more than just firewall info is going to be retrieved.